projects/epc/playground/src/proxies/hwui/controls/FadeIndicator.cpp

```diff
@@ -20,26 +20,46 @@ bool FadeIndicator::redraw(FrameBuffer& fb)
 {
   if(isVisible())
   {
-
     auto y = getPosition().getTop();
-    for(auto x = getPosition().getX(); x < getPosition().getX() + getWidth(); x++)
-    {
-      if(x > getPosition().getX() + (getPosition().getWidth() / 2))
+
+    if(m_voiceGroup == VoiceGroup::II)
```

Comment:
I'd like to suggest something (in my opinion) more readable (hopefully I got the intention right):

```suggestion
if(isVisible())
{
  auto y = getPosition().getTop();

  if(m_voiceGroup == VoiceGroup::II)
    y = getPosition().getBottom() + 1;

  auto center = getPosition().getCenter().getX();
  
  for(auto x = getPosition().getLeft(); x < getPosition().getRight(); x++)
  {
    if(m_voiceGroup == VoiceGroup::I && x > center)
      y++;

    if(m_voiceGroup == VoiceGroup::II && x < center)
      y--;

    drawColumn(fb, x, y);
  }
}
return true;
```

Reply:
agreed, and stole your suggestion